internal/resources/fetcher.go

```diff
@@ -148,16 +148,21 @@ func (f *Fetcher) GetClusterWideResourcesForPolicies(ctx context.Context, polici
 }
 
 func (f *Fetcher) getResourcesDynamically(ctx context.Context,
-	group string, version string, resource string, namespace string) (
+	resourceFilter *resourceFilter,
+	namespace string) (
 	*unstructured.UnstructuredList, error) {
 	resourceID := schema.GroupVersionResource{
-		Group:    group,
-		Version:  version,
-		Resource: resource,
+		Group:    resourceFilter.groupVersionResource.Group,
+		Version:  resourceFilter.groupVersionResource.Version,
+		Resource: resourceFilter.groupVersionResource.Resource,
 	}
 	var list *unstructured.UnstructuredList
 	var err error
-	list, err = f.dynamicClient.Resource(resourceID).Namespace(namespace).List(ctx, metav1.ListOptions{})
+	labelSelector := ""
```

Comment:
I'm not sure if this is necessary. The default value is [everything](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ListOptions) for the both fields, `LabelSelector` and `FieldSelector`. Considering the docs, I found weird why you need to define `fields.Everything()`. Because it should be the default value... :thinking: 

```
type ListOptions struct {
	[TypeMeta](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#TypeMeta) `json:",inline"`

	// A selector to restrict the list of returned objects by their labels.
	// Defaults to everything.
	// +optional
	LabelSelector [string](https://pkg.go.dev/builtin#string) `json:"labelSelector,omitempty" protobuf:"bytes,1,opt,name=labelSelector"`
	// A selector to restrict the list of returned objects by their fields.
	// Defaults to everything.
	// +optional
	FieldSelector [string](https://pkg.go.dev/builtin#string) `json:"fieldSelector,omitempty" protobuf:"bytes,2,opt,name=fieldSelector"`
```

Reply:
Anyway, I'll improve the code to avoid any corner case that it may faced if the label selector is not defined.

Reply:
reading that, yes, you are right!
Thanks for the improvements though!